File: GuessWhoOnePiece.UI/AnswerCard/AnswerCard.razor.cs
// <copyright file="AnswerCard.razor.cs">
// Copyright (c) 2025 All Rights Reserved. 
// </copyright>
// <author>Gabriel Marquette</author>

using GuessWhoOnePiece.Model;
using Microsoft.AspNetCore.Components;

namespace GuessWhoOnePiece.UI.AnswerCard;

public partial class AnswerCard : ComponentBase
{
    [Parameter] public string? AnswerText { get; set; }
    [Parameter] public string? AnswerImage { get; set; }
    [Parameter] public AnswerState? AnswerState { get; set; }
    
    private string DefineColor()
    {
        if (!string.IsNullOrEmpty(AnswerImage) && string.IsNullOrEmpty(AnswerText))
            return "answer-no-answer";

        return AnswerState switch
        {
            Model.AnswerState.Correct => "answer-correct",
            Model.AnswerState.Wrong => "answer-wrong",
            Model.AnswerState.NotAnswered => "answer-not-answered",
            _ => "answer-no-answer"
        };
    }
}

File: GuessWhoOnePiece.UI/AnswerCard/AnswerCard.razor.js
/* 
   Copyright (c) 2025 All Rights Reserved. 
   Author: Gabriel Marquette
*/

export function adjustFontSize() {
    const answerCircles = document.querySelectorAll(".answer-circle");

    answerCircles.forEach((circle) => {
        const text = circle.querySelector(".answer-text");
        if (text) {
            let fontSize = 40;
            text.style.fontSize = `${fontSize}px`;

            while (text.scrollHeight > circle.clientHeight && fontSize > 30) {
                fontSize--;
                text.style.fontSize = `${fontSize}px`;
            }
        }
    });
}